ClientExecutionPoolSizeLowTest Failures

**testNodeTerminateWithAsyncOperation**

```
java.lang.AssertionError: the size of the map is not correct expected:<1000> but was:<717>
    at org.junit.Assert.fail(Assert.java:88)
    at org.junit.Assert.failNotEquals(Assert.java:834)
    at org.junit.Assert.assertEquals(Assert.java:645)
    at com.hazelcast.test.HazelcastTestSupport$7.run(HazelcastTestSupport.java:667)
    at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:835)
    at com.hazelcast.test.HazelcastTestSupport.assertSizeEventually(HazelcastTestSupport.java:664)
    at com.hazelcast.test.HazelcastTestSupport.assertSizeEventually(HazelcastTestSupport.java:660)
    at com.hazelcast.client.io.ClientExecutionPoolSizeLowTest.testNodeTerminateWithAsyncOperations(ClientExecutionPoolSizeLowTest.java:80)
```

**testOwnerNodeTerminate**

```
com.hazelcast.core.HazelcastException: java.io.IOException: No available connection to address Address[127.0.0.1]:5701
    at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.getOrTriggerConnect(ClientConnectionManagerImpl.java:244)
    at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.getOrTriggerConnect(ClientConnectionManagerImpl.java:67)
    at com.hazelcast.client.spi.impl.ClientSmartInvocationServiceImpl.getConnection(ClientSmartInvocationServiceImpl.java:78)
    at com.hazelcast.client.spi.impl.ClientSmartInvocationServiceImpl.invokeOnPartitionOwner(ClientSmartInvocationServiceImpl.java:49)
    at com.hazelcast.client.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:152)
    at com.hazelcast.client.spi.impl.ClientInvocation.invoke(ClientInvocation.java:132)
    at com.hazelcast.client.spi.impl.ClientInvocation.run(ClientInvocation.java:163)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:67)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
    at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
    at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:92)
    at ------ End remote and begin local stack-trace ------.(Unknown Source)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveResponse(ClientInvocationFuture.java:133)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:100)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:75)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:38)
    at com.hazelcast.client.spi.ClientProxy.invokeOnPartition(ClientProxy.java:126)
    at com.hazelcast.client.spi.ClientProxy.invoke(ClientProxy.java:120)
    at com.hazelcast.client.proxy.ClientMapProxy.putInternal(ClientMapProxy.java:411)
    at com.hazelcast.client.proxy.ClientMapProxy.put(ClientMapProxy.java:405)
    at com.hazelcast.client.proxy.ClientMapProxy.put(ClientMapProxy.java:233)
    at com.hazelcast.client.io.ClientExecutionPoolSizeLowTest.testOwnerNodeTerminate(ClientExecutionPoolSizeLowTest.java:64)
```

**testNodeTerminate**

```
com.hazelcast.core.HazelcastException: java.io.IOException: No available connection to address Address[127.0.0.1]:5702
    at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.getOrTriggerConnect(ClientConnectionManagerImpl.java:244)
    at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl.getOrTriggerConnect(ClientConnectionManagerImpl.java:67)
    at com.hazelcast.client.spi.impl.ClientSmartInvocationServiceImpl.getConnection(ClientSmartInvocationServiceImpl.java:78)
    at com.hazelcast.client.spi.impl.ClientSmartInvocationServiceImpl.invokeOnPartitionOwner(ClientSmartInvocationServiceImpl.java:49)
    at com.hazelcast.client.spi.impl.ClientInvocation.invokeOnSelection(ClientInvocation.java:152)
    at com.hazelcast.client.spi.impl.ClientInvocation.invoke(ClientInvocation.java:132)
    at com.hazelcast.client.spi.impl.ClientInvocation.run(ClientInvocation.java:163)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at com.hazelcast.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:67)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471)
    at java.util.concurrent.FutureTask.run(FutureTask.java:262)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
    at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
    at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:92)
    at ------ End remote and begin local stack-trace ------.(Unknown Source)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.resolveResponse(ClientInvocationFuture.java:133)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:100)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:75)
    at com.hazelcast.client.spi.impl.ClientInvocationFuture.get(ClientInvocationFuture.java:38)
    at com.hazelcast.client.spi.ClientProxy.invokeOnPartition(ClientProxy.java:126)
    at com.hazelcast.client.spi.ClientProxy.invoke(ClientProxy.java:120)
    at com.hazelcast.client.proxy.ClientMapProxy.putInternal(ClientMapProxy.java:411)
    at com.hazelcast.client.proxy.ClientMapProxy.put(ClientMapProxy.java:405)
    at com.hazelcast.client.proxy.ClientMapProxy.put(ClientMapProxy.java:233)
    at com.hazelcast.client.io.ClientExecutionPoolSizeLowTest.testNodeTerminate(ClientExecutionPoolSizeLowTest.java:53)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-nightly/807/#showFailuresLink

--- a/hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java
+++ b/hazelcast-client-legacy/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java
@@ -235,6 +235,7 @@ public final class ClientPartitionServiceImpl implements ClientPartitionService 
 
             Connection connection = getOwnerConnection();
             if (connection == null) {
+                updating.set(false);
                 return;
             }
             ClientInvocationFuture clientInvocationFuture = getPartitionsFrom(connection);

--- a/hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java
+++ b/hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientPartitionServiceImpl.java
@@ -238,6 +238,7 @@ public final class ClientPartitionServiceImpl
 
             Connection connection = getOwnerConnection();
             if (connection == null) {
+                updating.set(false);
                 return;
             }
             ClientInvocationFuture clientInvocationFuture = getPartitionsFrom(connection);

